dask/dataframe/io/parquet/core.py

```diff
@@ -852,14 +859,14 @@ def create_metadata_file(
 
     # Convert the raw graph to a `Delayed` object
     graph = HighLevelGraph.from_collections(name, dsk, dependencies=[])
-    out = Delayed(name, graph)
 
     # Optionally compute the result
     if compute:
         if compute_kwargs is None:
             compute_kwargs = dict()
-        out = out.compute(**compute_kwargs)
-    return out
+        return compute_as_if_collection(DataFrame, graph, name, **compute_kwargs)
+    else:
+        return Delayed(name, graph)
```

Comment:
Thanks @mrocklin, overall this looks good. I'm wondering if this change is necessary. I don't think there are any Dask DataFrames involved with `create_metadata_file`, so I could think that the normal Delayed optimizations (i.e. culling) would be the best we could do. Am I missing something?

Reply:
Ah, that could be.  I went through this file quickly